pkg/eraser/eraser.go

```diff
@@ -90,64 +129,32 @@ func getImageClient(ctx context.Context) (pb.ImageServiceClient, *grpc.ClientCon
 	return imageClient, conn, nil
 }
 
-func listImages(ctx context.Context, client pb.ImageServiceClient, image string) (resp *pb.ListImagesResponse, err error) {
-	request := &pb.ListImagesRequest{Filter: &pb.ImageFilter{Image: &pb.ImageSpec{Image: image}}}
-
-	resp, err = client.ListImages(ctx, request)
-	if err != nil {
-		return nil, err
-	}
-
-	return resp, nil
-}
-
-func removeImage(ctx context.Context, client pb.ImageServiceClient, image string) (resp *pb.RemoveImageResponse, err error) {
-	if image == "" {
-		return nil, err
-	}
-
-	request := &pb.RemoveImageRequest{Image: &pb.ImageSpec{Image: image}}
-
-	resp, err = client.RemoveImage(ctx, request)
-	if err != nil {
-		return nil, err
-	}
-
-	return resp, nil
-}
-
-func removeVulnerableImages() (err error) {
+func removeVulnerableImages(c *client, socketPath string, imagelistName string) (err error) {
```

Comment:
Just noting from our discussion, this needs the client interface we discussed and not a `*client`.